src/app/places.service.ts

```diff
@@ -1,32 +1,103 @@
-import { Injectable} from '@angular/core';
+import { Injectable } from '@angular/core';
 import data from './data/places-katowice.json';
 import { Place } from '../models/place';
 import { RangeInterface } from './slider/slider.component';
 import { SearchInterface } from './search/search.component';
-
+import { SelectedOptionsInterface } from './side-bar/side-bar.component';
+import { Router } from '@angular/router';
 
 @Injectable({
   providedIn: 'root',
 })
 export class PlacesService {
   private places: Place[] = data;
 
-  constructor() {}
+  constructor(private router: Router) {}
+
   getPlaces(): Place[] {
+    return data;
+  }
+
+  filterPlaces(selectedOptions: SelectedOptionsInterface): Place[] {
+    this.places = data;
+
+    if (selectedOptions.searchPlaces) {
+      this.getFilteredPlacesByName({
+        searchText: selectedOptions.searchPlaces,
+      });
+    }
+
+    if (selectedOptions.filterPlacesByRating) {
+      this.getFilteredPlacesByRating({
+        from: selectedOptions.filterPlacesByRating.from,
+        to: selectedOptions.filterPlacesByRating.to,
+      });
+    }
+
+    if (selectedOptions.sortPlaces) {
+      this.sortPlaces(selectedOptions.sortPlaces);
+    }
+
     return this.places;
   }
 
-  getFilteredPlacesByRating(options: RangeInterface): Place[] {
-    return this.places.filter((place: Place) => {
+  sortPlaces(selectedOption: string) {
+    if (selectedOption === 'No-sort') {
```

Comment:
It would be nice to create ENUM with this options to be used here and in sorting component. To avoid typos and minutes of debugging in code :)